tests/pipeline/test_numerical_expression.py

```diff
@@ -147,6 +147,41 @@ def test_validate_bad(self):
         with self.assertRaises(TypeError):
             (f > f) > f
 
+    def test_many_inputs(self):
+        factors = []
+
+        # Create 11 factors to use as inputs. We need at least 11 to create a
+        # string expression containing an x_10. Then, ensure that this long
+        # expression can correctly merge with another.
+        for class_name in 'PQRSTUVWXYZ':
+            factors.append(
+                type(
+                    class_name,
+                    (Factor,),
+                    dict(dtype=float64_dtype, inputs=(), window_length=0),
+                )
+            )
+
+        short_num_expr = NumericalExpression(
```

Comment:
I don't think there's a benefit to manually constructing these.  We can just use '+' or similar.